x-pack/plugins/fleet/public/applications/fleet/sections/epm/components/package_list_grid.tsx

```diff
@@ -143,3 +163,54 @@ function GridColumn({ list }: GridColumnProps) {
     </EuiFlexGrid>
   );
 }
+
+interface MissingIntegrationContentProps {
+  resetQuery: () => void;
+  setSelectedCategory: (category: string) => void;
+}
+
+function MissingIntegrationContent({
+  resetQuery,
+  setSelectedCategory,
+}: MissingIntegrationContentProps) {
+  const history = useHistory();
+  const { getPath } = useLink();
+
+  const handleCustomInputsLinkClick = useCallback(() => {
+    resetQuery();
+    setSelectedCategory('custom');
+
+    history.push({
+      pathname: getPath('integrations_all'),
```

Comment:
I think a better solution here might be to use URL parameters to support fully deep linking into this integrations search, e.g. 

```
http://localhost:5601/app/fleet#/integrations?query=hello&category=custom
```

but in trying that things quickly got a little in the weeds. We need to hoist up the search term + category state and initialize them from URL parameters and that implementation started to feel a bit clunky. 

Happy to hear input if anyone has any suggestions for making this behave like a standard link instead of the way it is now.

Reply:
This also brings about the question of whether the search for "installed integration" and "all integrations" should share the same URL-driven state or not.